data.md

```diff
@@ -872,15 +872,15 @@ module EVM-DATA-SYMBOLIC [symbolic]
       => #bufSeg(BUF, START0 +Int START, WIDTH)
       requires 0 <=Int START andBool START +Int WIDTH <=Int WIDTH0
 
-    rule #take(N, BUF      ) => #take(N, BUF ++ .WordStack)               requires #isBuf(BUF)
+    rule #take(N, BUF      ) => #bufSeg(BUF, 0, N)                        requires #isBuf(BUF) andBool 0 <=Int N andBool N <=Int #sizeBuffer(BUF)
```

Comment:
this is not complete. if N > #sizeBuffer, it should have the zero padding to the right.